engine-rest/engine-rest/src/main/java/org/camunda/bpm/engine/rest/spi/impl/AbstractProcessEngineAware.java

```diff
@@ -31,5 +33,10 @@ public AbstractProcessEngineAware(String engineName) {
     processEngine = EngineUtil.lookupProcessEngine(engineName);
   }
 
-
+  protected ProcessEngine getProcessEngine() {
+    if (processEngine == null) {
+      throw new InvalidRequestException(Status.BAD_REQUEST, "No process engine available");
```

Comment:
Simply by name it is implied that the subclass ``AbstractRestProcessEngineAware``  is the class explicitly handling all REST concerns, while ``AbstractProcessEngineAware`` is a more general class. Therefore, I'm not quite comfortable with throwing an explicit REST exception in the more general ``AbstractProcessEngineAware``.
However, as ``AbstractRestProcessEngineAware`` is its only subclass (as far as I have been able to find any), I would suggest the following:   
Completly remove ``AbstractProcessEngineAware`` and move its functionality into ``AbstractRestProcessEngineAware``. That way, ``processEngine`` can still be private (but in ``AbstractRestProcessEngineAware``), will never be accessed directly, but REST exceptions are still only thrown in classes that are actually concerned with offering REST services.

Reply:
I'll look into why we have this separation in the first place and if we can merge this hierarchy. In general, it would be fine to leave it as is as well since we are still talking about the abstract superclass inside the REST API project. So, throwing REST exceptions here should still be ok.

Reply:
I am not sure about removing `AbstractProcessEngineAware`. At the moment, when creating new rest service implementation class that extends `AbstractRestProcessEngineAware`, the constructor with `objectMapper` must be created. The latter is used for rest service initializer. The implementation won't change in case of the refactoring. I just wonder that might better to keep hierarchy due to that explicit constructor that we use.
`AbstractProcessEngineAware` is located in `spi.impl` package, no idea why and I do not see restrictions to remove it because of that.
I cannot think of other cons of the refactoring, the pros are clear.

Reply:
Merged `AbstractProcessEngineAware` into `AbstractRestProcessEngineAware` with 9f0cd95555875349195079fcf64e902ad659824b. Let's see what the CI says about this. I cannot find any specific reason to keep `AbstractProcessEngineAware`.